internal/pkg/actions/action.go

```diff
@@ -0,0 +1,160 @@
+package actions
+
+import (
+	"bytes"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	updateOnChangeAnnotation = "reloader.stakater.com/update-on-change"
+)
+
+// Action interface so that other actions like slack can implement this
+type Action interface {
+	ObjectCreated(obj interface{}, client kubernetes.Interface)
+	ObjectDeleted(obj interface{})
+	ObjectUpdated(oldObj interface{}, client kubernetes.Interface)
+}
+
+// Default class with empty implementations for any action that we dont support currently
+type Default struct {
+}
+
+// ObjectCreated Do nothing for default handler
+func (d *Default) ObjectCreated(obj interface{}, client kubernetes.Interface) {
+	message := "Configmap: `" + obj.(*v1.ConfigMap).Name + "`has been created in Namespace: `" + obj.(*v1.ConfigMap).Namespace + "`"
+	logrus.Infof(message)
+	err := rollingUpgradeDeployments(obj, client)
+	if err != nil {
+		logrus.Errorf("failed to update Deployment: %v", err)
+	}
+}
+
+// ObjectDeleted Do nothing for default handler
+func (d *Default) ObjectDeleted(obj interface{}) {
+
+}
+
+// ObjectUpdated Do nothing for default handler
+func (d *Default) ObjectUpdated(oldObj interface{}, client kubernetes.Interface) {
+	message := "Configmap: `" + oldObj.(*v1.ConfigMap).Name + "`has been updated in Namespace: `" + oldObj.(*v1.ConfigMap).Namespace + "`"
+	logrus.Infof(message)
+	err := rollingUpgradeDeployments(oldObj, client)
+	if err != nil {
+		logrus.Errorf("failed to update Deployment: %v", err)
+	}
+}
+
+// Implementation has been borrowed from fabric8io/configmapcontroller
+// Method has been modified a little to use updated liberaries.
+func rollingUpgradeDeployments(oldObj interface{}, client kubernetes.Interface) error {
+	ns := oldObj.(*v1.ConfigMap).Namespace
+	configMapName := oldObj.(*v1.ConfigMap).Name
+	configMapVersion := convertConfigMapToToken(oldObj.(*v1.ConfigMap))
+
+	deployments, err := client.Apps().Deployments(ns).List(meta_v1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to list deployments")
+	}
+	for _, d := range deployments.Items {
+		containers := d.Spec.Template.Spec.Containers
+		// match deployments with the correct annotation
+		annotationValue, _ := d.ObjectMeta.Annotations[updateOnChangeAnnotation]
```

Comment:
should write annotationValue := d.ObjectMeta.Annotations[updateOnChangeAnnotation] instead of annotationValue, _ := d.ObjectMeta.Annotations[updateOnChangeAnnotation]